test/junit/scala/collection/IteratorTest.scala

```diff
@@ -600,4 +600,15 @@ class IteratorTest {
       assertEquals(1, i)
     }
   }
+
+  @Test
+  def t11106(): Unit = {
+    var i = 0
+    Iterator.continually(0)
+      .map(_ => {i += 1; i})
+      .withFilter(_ < 10)
+      .take(3)
+      .foreach(_ => ())
```

Comment:
For me, it's easier to see with 
```
scala 2.13.0-M4> var i = 0; Iterator.continually { i += 1; i }.withFilter(_ => true).take(3).toList
i: Int = 4
res0: List[Int] = List(1, 2, 3)
```
where both `i` and the length should be 3.

Reply:
I have no strong preference; I just copied from the bug ticket.

What you suggest seems fine to me. Feel free to make the change and force-push if you want - I won't be able to do get to it for a few days at least.